encfs/BlockFileIO.cpp

```diff
@@ -256,7 +256,7 @@ ssize_t BlockFileIO::write(const IORequest &req) {
     // if writing an entire block, or writing a partial block that requires
     // no merging with existing data..
     if ((toCopy == _blockSize) ||
-        (partialOffset == 0 && blockReq.offset + toCopy >= fileSize)) {
+        (partialOffset == 0 && blockReq.offset + (off_t)toCopy >= fileSize)) {
```

Comment:
toCopy is not signed, is it?

Reply:
No, so without the cast this leads to a comparison between size_t and off_t :
`comparison of integers of different signs: 'unsigned long' and 'off_t' (aka 'long') [clang-diagnostic-sign-compare]`